### PR TITLE
Protection for extrapolation of laser corrections (reload PR #30254)

### PR DESCRIPTION
**Pull request by fcouderc**
reload of PR #30254 

#### PR description:
add extra parameter to protect for extrapolation of laser corrections.
 
if t > t3 + dt : t = t3+dt

#### PR validation:

addOnTests.py
runTheMatrix.py was not re-run, but was working perfectly for the last PR 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by christopheralanwest** [comment]
please test

@fcouderc In case you were unaware, you may find the [instructions for rebasing and porting features](http://cms-sw.github.io/tutorial-resolve-conflicts.html) useful.

**Comment by fcouderc** [comment]
All failures explained in PR#30254.

**Comment by fcouderc** [comment]
FYI @Martin-Grunewald

**Comment by Martin-Grunewald** [comment]
Thanks! And no HLT signature needed anymore.

File: CalibCalorimetry/EcalLaserCorrection/python/ecalLaserCorrectionService_cfi.py [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import FWCore.ParameterSet.Config as cms
 
-EcalLaserCorrectionService = cms.ESProducer("EcalLaserCorrectionService")
+EcalLaserCorrectionService = cms.ESProducer("EcalLaserCorrectionService", 
+                                        ### delta t is the safety margin before stopping
+                                        ### the laser correction extrapolation (this is in seconds)
+                                        maxExtrapolationTimeInSec = cms.uint32( 0 ) )
+
```
Comment (Martin-Grunewald) [inline CalibCalorimetry/EcalLaserCorrection/python/ecalLaserCorrectionService_cfi.py:8]: Hi,

Now with fillDescriptions, you no longer need the explicit cfi file in the ../python directory. Upon compilation of the c++ code with scram, a cfi file will be generated from fillDescriptions and placed into ../cfipython, accessible thru the same path  CalibCalorimetry.EcalLaserCorrection.EcalLaserCorrectionService_cfi

**Comment by christopheralanwest** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Comment by fcouderc** [comment]
FYI @blenzi

**Pull request merged**